widgetssdk/src/main/java/com/glia/widgets/core/callvisualizer/CallVisualizerManager.java

```diff
@@ -44,16 +44,22 @@ private void checkForProperInit() {
 
     @Override
     public VisitorCodeView createVisitorCodeView(Context context) {
-        checkForProperInit();
-        startListeningForEngagements();
-        return buildVisitorCodeUseCase.invoke(context, false);
+        if (!engagementRepository.isCallVisualizerEngagement()) {
+            checkForProperInit();
+            startListeningForEngagements();
+            return buildVisitorCodeUseCase.invoke(context, false);
+        } else {
+            return null;
+        }
```

Comment:
This is a custom view. I don't think it still should continue working as before. In my opinion this is already the Integrator responsibility to handle case when there is already ongoing engagement

Reply:
Yes, but we need to decide if we are allowing this approach or not. Returning a null here will avoid the integrator from using our sdk in a way that we do not approve.

Reply:
I think that this makes things more complicated but does not really prevent the integrator from showing the visitor code during engagement since if he created the custom Visitor Code view before the engagement that view would still continue to get displayed and refreshed. So returning the null does not really add benefit but if you do that you will need to still add extra explanations to the integrator guides saying that this method might also return 'null' and why it does that.

But if you still think this way is better then I just ask you to update the integration guides.